modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaHomogenousCdsIndexTradePricer.java

```diff
@@ -0,0 +1,304 @@
+   * 
+   * @param trade  the trade
+   * @param ratesProvider  the rates provider
+   * @param refData  the reference data
+   * @return the present value sensitivity
+   */
+  public PointSensitivities presentValueOnSettleSensitivity(
+      ResolvedCdsIndexTrade trade,
+      CreditRatesProvider ratesProvider,
+      ReferenceData refData) {
+
+    LocalDate settlementDate = calculateSettlementDate(trade, ratesProvider, refData);
+    return productPricer.presentValueSensitivity(trade.getProduct(), ratesProvider, settlementDate, refData).build();
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the risky PV01 of the underlying product. 
+   * <p>
+   * RPV01 is defined as minus of the present value sensitivity to coupon rate.
+   * <p>
+   * This is computed based on the settlement date rather than the valuation date.
+   * 
+   * @param trade  the trade
+   * @param ratesProvider  the rates provider
+   * @param priceType  the price type
+   * @param refData  the reference date
+   * @return the RPV01
+   */
+  public CurrencyAmount rpv01OnSettle(
+      ResolvedCdsIndexTrade trade,
+      CreditRatesProvider ratesProvider,
+      PriceType priceType,
+      ReferenceData refData) {
+
+    LocalDate settlementDate = calculateSettlementDate(trade, ratesProvider, refData);
+    return productPricer.rpv01(trade.getProduct(), ratesProvider, settlementDate, priceType, refData);
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the recovery01 of the underlying product.
+   * <p>
+   * The recovery01 is defined as the present value sensitivity to the recovery rate.
+   * Since the ISDA standard model requires the recovery rate to be constant throughout the lifetime of the CDS,  
+   * one currency amount is returned by this method.
+   * <p>
+   * This is computed based on the settlement date rather than the valuation date.
+   * 
+   * @param trade  the trade
+   * @param ratesProvider  the rates provider
+   * @param refData  the reference data
+   * @return the recovery01
+   */
+  public CurrencyAmount recovery01OnSettle(
+      ResolvedCdsIndexTrade trade,
+      CreditRatesProvider ratesProvider,
+      ReferenceData refData) {
+
+    LocalDate settlementDate = calculateSettlementDate(trade, ratesProvider, refData);
+    return productPricer.recovery01(trade.getProduct(), ratesProvider, settlementDate, refData);
+  }
+
+  //-------------------------------------------------------------------------
+  private LocalDate calculateSettlementDate(
+      ResolvedCdsIndexTrade trade,
+      CreditRatesProvider ratesProvider,
+      ReferenceData refData) {
+
+    return trade.getInfo().getSettlementDate()
+        .orElse(trade.getProduct().getSettlementDateOffset().adjust(ratesProvider.getValuationDate(), refData));
```

Comment:
Should add a method `calculateSettlementDate` on `ResolvedCdsIndex` that takes the valuation date and reference data. Like `calculateSettlementDateFromValuation` on `ResolvedCapitalIndexedBond`.